server/indexer/indexer.go

```diff
@@ -0,0 +1,301 @@
+			return fmt.Errorf("something wrong at iterFn: %v", err)
+		}
+		if (tile.Children != nil) && len(*tile.Children) != 0 {
+			for _, child := range *tile.Children {
+				err = iterTile(&child, computedTransform)
+				if err != nil {
+					return fmt.Errorf("something went wrong at iterTile: %v", err)
+				}
+			}
+		}
+		return nil
+	}
+
+	err := iterTile(root, eyeMat(4))
+	if err != nil {
+		return fmt.Errorf("something went wrong at iterTile: %v", err)
+	}
+
+	return nil
+}
+
+func ReadTilesetFeatures(ts *tiles.Tileset, indexesConfig IndexesConfig, tilesetDir, outDir string) (map[string]TilesetFeature, error) {
+	uniqueFeatures := make(map[string]TilesetFeature)
+	tilesetQueue := []*tiles.Tileset{ts}
+
+	for _, tileset := range tilesetQueue {
+
+		tilesetIterFn := func(tile *tiles.Tile, computedTransform *mat.Dense) error {
+			tileUri, err := tile.Uri()
+			if err != nil {
+				return fmt.Errorf("failed to fetch uri of tile: %v", err)
+			}
+			contentPath := filepath.Join(tilesetDir, tileUri)
+			fmt.Println(contentPath)
+
+			if strings.HasSuffix(contentPath, ".json") {
+				childTileset, _ := tiles.Open(contentPath)
+				tilesetQueue = append(tilesetQueue, childTileset)
+				return nil
+			}
+			b3dm, err := b3dms.Open(contentPath)
+			if err != nil {
+				return fmt.Errorf("failed to open b3dm file: %v", err)
+			}
+			featureTable := b3dm.GetFeatureTable()
+			batchLength := featureTable.GetBatchLength()
+			featureTableView := b3dm.GetFeatureTableView()
+			batchTable := b3dm.GetBatchTable()
+			batchTableProperties := batchTable.Data
+			computedFeaturePositions := []Cartographic{}
+			gltf := b3dm.GetModel()
+			if gltf != nil {
+				rtcTransform, err := getRtcTransform(featureTableView, gltf)
+				if err != nil {
+					return fmt.Errorf("failed to getRtcTransform: %v", err)
+				}
+				toZUpTransform := getZUpTransform()
+				computedFeaturePositions, err = computeFeaturePositionsFromGltfVertices(
+					gltf,
+					computedTransform,
+					rtcTransform,
+					toZUpTransform,
+					batchLength,
+				)
+				if err != nil {
+					return fmt.Errorf("failed to open b3dm file: %v", err)
+				}
+			}
+
+			for batchId := 0; batchId < batchLength; batchId++ {
+				batchProperties := make(map[string]interface{})
+				for name, values := range batchTableProperties {
+					batchProperties[name] = nil
+					if len(values) > 0 {
+						batchProperties[name] = values[batchId]
+					}
+				}
+				position := computedFeaturePositions[batchId]
+				idValue := batchProperties[indexesConfig.IdProperty].(string)
+				uniqueFeatures[idValue] = TilesetFeature{
+					Position:   position,
+					Properties: batchProperties,
+				}
+			}
+
+			return nil
+		}
+		err := ForEachTile(tileset, tilesetIterFn)
+		if err != nil {
+			return nil, fmt.Errorf("something went wrong at iterTile: %v", err)
+		}
+	}
+
+	return uniqueFeatures, nil
+
+}
+
+func Indexer(tilesetFile, indexConfigFile, outDir string) error {
+	tileset, err := tiles.Open(tilesetFile)
```

Comment:
```suggestion
	tileset, err := tiles.Open("tileset.json") // or config.TilesetJSONPath
```

I believe the specification specifies that the file name is `tileset.json`, but if you need flexibility, you should be able to set that in `IndexerConfig`.